### DISCUSSION TITLE
Approach to filtering only today's date

### DISCUSSION THREAD

**Discussion by gmcamposano** [prisma/prisma#16128] [Q&A] [answered]
Can someone please suggest if this is a good approach of if there is a better one for only filtering todays created_at only rows:

```
const res = await prisma.facturas.findMany({
    where: {
      created_at: {
        gte: new Date(new Date().setHours(0, 0, 0, 0)),
        lte: new Date(new Date().setHours(23, 59, 59, 999))
      }
    }
  })
```

**Comment by nurul3101** [comment, marked as answer]
Hey @gmcamposano 👋 
The approach that you have suggested should work. You can also do something like this if you have access to the date value:
```
  const data = await prisma.facturas.findMany({
    where: {
      createdAt: {
        gte: new Date("2022-01-01"),
        lt:  new Date("2022-01-02")
      },
    },
  });
```

**Reply by gmcamposano** [reply]
Thanks for answering. I made a mistake:
`new Date(new Date('2022-11-09').setHours(0, 0, 0, 0))` //Tue Nov 08 2022 00:00:00 GMT-0400 (Atlantic Standard Time)

I have an input with only one date, as the users is only going to lookup that day

As you can see I am GMT -4 so, what I am trying to accomplish is:
If I pass a date like '2022-11-09' it filters taking in consideration only that day which in turn will be from: '2022-11-09T00:00:000Z  to: '2022-11-09:23:59:999Z'.

**Reply by pantharshit00** [reply]
Do you want to use `UTC` or `GMT-4` while searching? It is not clear to me with the above message. If you want `GMT-4` then your method should work as it will correctly convert time to UTC from your timezone. You might also want to look into a date library like date-fns(https://date-fns.org/) where you can do something like this:
```ts
const gte = zonedTimeToUtc('2022-11-09', 'America/Puerto_Rico') // 2022-11-09T04:00:00.000Z
cost le = zonedTimeToUtc('2022-11-09 23:59:59.999', 'America/Puerto_Rico') // 2022-11-10T03:59:59.999Z
```

**Reply by gmcamposano** [reply]
Hi @pantharshit00 
Sorry for the confusion I have created. Correct, I am GMT-4 and `new Date('2022-11-11') //Thu Nov 10 2022 20:00:00 GMT-0400 (Atlantic Standard Time) .` I then realized that new Date() takes into consideration the timezone
I was trying to achieve adding one day to the selected date so that I can filter one whole day. As to now I have the following expresion to make this happen:  `new Date(hasta.setDate(hasta.getDate() + 1))` which I dont know if its correct. 
```
const data = await prisma.facturas.findMany({
    where: {
      createdAt: {
        gte: new Date(date),     //2022-11-10
        lt:  xxxxxxxx     <=== date + 1 day    //2022-11-11
      },
    },
  });
```

**Reply by nurul3101** [reply]
@gmcamposano 👋 
You are adding one day to the existing date and applying the filter which looks good. Are you facing any issues now?

**Reply by vladi-stevanovic** [reply]
👋 Hi @gmcamposano are you still experiencing an issue here? Please let us know so we can continue to investigate!

Or, if you've found a solution, we'd love to know how you solved this problem. Feel free to post it in this thread and "Mark as answer" if this discussion is now solved!

**Reply by gmcamposano** [reply]
@vladi-stevanovic sorry about that. @nurul3101 Its all good now. Thank you

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
